data-structures/lists/python/astronaut_list_2D.py

```diff
@@ -0,0 +1,284 @@
+        # Repeat for each value in the inner 1D list
+        for value in person:
+            print(f"{value}  ", end = "") # Do not print new line
+        print() # Print new line
+
+
+def add_new_astronaut(astronauts, given_craft, given_name, given_days):
+    """Append a new astronaut list to the end of the 2D list"""
+
+    # Create a new 1D list for the astronaut
+    new_astronaut = [given_craft, given_name, given_days]
+
+    # Append the new astronaut list to the end of the 2D list
+    astronauts.append(new_astronaut)
+
+    return astronauts
+
+
+def add_new_value_to_astronaut(astronauts, astronaut_index, new_value):
+    """Append a new value to the end of an astronaut list"""
+
+    # Append the new value to the astronaut list at the given index
+    astronauts[astronaut_index].append(new_value)
+
+    return astronauts
+
+
+# Remove validation for the platform version
+def update_days_in_space(astronauts, astronaut_index, new_days):
+    """Update the number of days in space for an astronaut"""
+
+    # Check the astronaut index is valid
+    if astronaut_index >= 0 and astronaut_index < len(astronauts):
+        astronauts[astronaut_index][DAYS_INDEX] = new_days
+    else:
+        print(f"Astronaut index {astronaut_index} out of range")
+
+    return astronauts
+
+
+# Think of a better function name?
+def find_astronaut_return_index(astronauts, search_name):
+    """Find an astronaut that contains the given name"""
+
+    # Return -1 if no astronaut found
+    found_index = -1
+
+    # Repeat for all astronauts in the 2D list using index i
+    for i in range(len(astronauts)):
+        # Substring search for the given name
+        if search_name in astronauts[i][NAME_INDEX]:
+            # Stop searching and return the index
+            found_index = i
+            break
+
+    return found_index
+
+
+# Think of a better function name?
+def find_astronaut_return_list(astronauts, search_name):
+    """Find an astronaut that contains the given name"""
+
+    # A 2D list to store each astronaut list that contains a name
+    found_list = []
+
+    # Repeat for all astronauts in the 2D list using index i
+    for i in range(len(astronauts)):
+        # Substring search for the given search name
+        if search_name in astronauts[i][NAME_INDEX]:
+            found_list.append(astronauts[i])
+
+    return found_list
+
+
+# Remove validation for the platform version
+def delete_astronaut(astronauts, astronaut_index):
+    """Delete an astronaut at the given index"""
+
+    # Check the astronaut index is valid
+    if astronaut_index >= 0 and astronaut_index < len(astronauts):
+        del astronauts[astronaut_index]
+    else:
+        print(f"Astronaut index {astronaut_index} out of range")
+
+    return astronauts
+
+
+# Remove validation for the platform version
+def delete_value_from_astronaut(astronauts, astronaut_index, value_index):
+    """Delete a value from an astronaut at the given indices"""
```

Comment:
I am ab bt confused as to why you would want to so this. Doesn't it make the constants go awry?